pkg/gadgets/trace/tcpdrop/tracer/bpf/tcpdrop.bpf.c

```diff
@@ -0,0 +1,110 @@
+// Copyright 2023 Microsoft Corporation
+
+#include <vmlinux/vmlinux.h>
+
+#include <bpf/bpf_helpers.h>
+#include <bpf/bpf_core_read.h>
+#include <bpf/bpf_tracing.h>
+#include <bpf/bpf_endian.h>
+
+#include "tcpdrop.h"
+
+/* Define here, because there are conflicts with include files */
+#define AF_INET		2
+#define AF_INET6	10
+
+// we need this to make sure the compiler doesn't remove our struct
+const struct event *unusedevent __attribute__((unused));
+
+struct {
+	__uint(type, BPF_MAP_TYPE_PERF_EVENT_ARRAY);
+	__uint(key_size, sizeof(__u32));
+	__uint(value_size, sizeof(__u32));
+} events SEC(".maps");
+
+// This struct is defined according to the following format file:
+// /sys/kernel/tracing/events/skb/kfree_skb/format
+struct kfree_skb_ctx {
+	/* The first 8 bytes is not allowed to read */
+	unsigned long pad;
+
+	void *skbaddr;
+	void *location;
+	unsigned short protocol;
+	int reason;
+};
+
+static __always_inline int __trace_tcp_drop(void *ctx, struct sock *sk, struct sk_buff *skb, int reason)
+{
+	if (sk == NULL)
+		return 0;
+
+	u_int8_t *tcphdr = (u_int8_t *)(BPF_CORE_READ(skb, head) + BPF_CORE_READ(skb, transport_header));
+	struct inet_sock *sockp = (struct inet_sock *)sk;
+	struct task_struct *task = (struct task_struct*) bpf_get_current_task();
+
+	struct event event = {};
+	event.timestamp = bpf_ktime_get_boot_ns();
+	event.af = BPF_CORE_READ(sk, __sk_common.skc_family);
+	event.state = BPF_CORE_READ(sk, __sk_common.skc_state);
+	event.reason = reason;
+	bpf_get_current_comm(&event.comm, sizeof(event.comm));
+	event.pid = bpf_get_current_pid_tgid() >> 32;
+	event.mntns_id = (u64) BPF_CORE_READ(task, nsproxy, mnt_ns, ns.inum);
+	bpf_probe_read_kernel(&event.tcpflags, sizeof(event.tcpflags), tcphdr+13);
+
+	BPF_CORE_READ_INTO(&event.dport, sk, __sk_common.skc_dport);
+	if (event.dport == 0)
+		return 0;
+
+	BPF_CORE_READ_INTO(&event.sport, sockp, inet_sport);
+	if (event.sport == 0)
+		return 0;
+
+	if (event.af == AF_INET) {
+		BPF_CORE_READ_INTO(&event.saddr_v4, sk, __sk_common.skc_rcv_saddr);
+		if (event.saddr_v4 == 0)
+			return 0;
+
+		BPF_CORE_READ_INTO(&event.daddr_v4, sk, __sk_common.skc_daddr);
+		if (event.daddr_v4 == 0)
+			return 0;
+
+	} else if (event.af == AF_INET6) {
+		bpf_probe_read_kernel(&event.daddr, sizeof(event.daddr),
+			sk->__sk_common.skc_v6_daddr.in6_u.u6_addr32);
+		bpf_probe_read_kernel(&event.saddr, sizeof(event.saddr),
+			sk->__sk_common.skc_v6_rcv_saddr.in6_u.u6_addr32);
+	} else {
+		// drop
+		return 0;
+	}
+
+	bpf_perf_event_output(ctx, &events, BPF_F_CURRENT_CPU, &event, sizeof(event));
+	return 0;
+}
+
+SEC("tracepoint/skb/kfree_skb")
+int ig_tcpdrop(struct kfree_skb_ctx *ctx)
+{
+	struct sk_buff *skb = ctx->skbaddr;
+	struct sock *sk = BPF_CORE_READ(skb, sk);
+	int reason = ctx->reason;
+
+	#define SKB_DROP_REASON_NOT_SPECIFIED 2
+	if (reason > SKB_DROP_REASON_NOT_SPECIFIED) {
+		return __trace_tcp_drop(ctx, sk, skb, reason);
+	}
```

Comment:
To keep it consistent? 

```suggestion
	if (reason > SKB_DROP_REASON_NOT_SPECIFIED) 
		return __trace_tcp_drop(ctx, sk, skb, reason);	
```